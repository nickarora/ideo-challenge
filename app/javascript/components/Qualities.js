import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { connect } from 'react-redux'
import FlipMove from 'react-flip-move'
import QualityPanel from './QualityPanel'
import sortQualities from '../util/sortQualities'

export const Container = styled.section`
  margin-top: 16px;
`

const Qualities = ({ sortConfig, qualities }) => (
  <Container>
    <FlipMove appearAnimation="fade">
      {sortQualities(qualities, sortConfig).map(quality => (
        <div key={quality.name} className="col-md-4">
          <QualityPanel quality={quality} />
        </div>
      ))}
    </FlipMove>
  </Container>
)

Qualities.propTypes = {
  sortConfig: PropTypes.shape({
    sortOrder: PropTypes.string.isRequired,
    scoreAscending: PropTypes.bool.isRequired,
    nameAscending: PropTypes.bool.isRequired,
  }).isRequired,
  qualities: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      color: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      score: PropTypes.number.isRequired,
    })
  ).isRequired,
}

const mapStateToProps = (state, _ownProps) => ({
  sortConfig: state.creativeQualitySortOrder,
})

export default connect(mapStateToProps)(Qualities)
