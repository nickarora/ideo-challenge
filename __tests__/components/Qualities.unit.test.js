import Qualities from '~/components/Qualities'

jest.mock('assets/images/qualityIcons/collaboration.png', () => 'img')
jest.mock('assets/images/qualityIcons/empowerment.png', () => 'img')
jest.mock('assets/images/qualityIcons/purpose.png', () => 'img')
jest.mock('~/util/sortQualities', () => arr => arr)
jest.mock('react-redux', () => ({
  connect: () => Component => props => (
    <Component
      {...props}
      sortConfig={{
        sortOrder: 'SCORE',
        scoreAscending: true,
        nameAscending: true,
      }}
    />
  ),
}))

const mockQualites = [
  {
    name: 'Foo',
    color: '#333',
    description: 'some description',
    score: 50,
  },
  {
    name: 'Bar',
    color: '#AAA',
    description: 'other description',
    score: -25,
  },
]

describe('Qualities', () => {
  it('renders a list of qualities', () => {
    const wrapper = shallow(<Qualities qualities={mockQualites} />).dive()
    expect(wrapper).toMatchSnapshot()
  })
})
