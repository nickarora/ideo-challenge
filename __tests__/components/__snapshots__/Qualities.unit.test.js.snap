// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Qualities renders a list of qualities 1`] = `
<styled.section>
  <FlipMovePropConverter
    delay={0}
    disableAllAnimations={false}
    duration={350}
    easing="ease-in-out"
    enterAnimation="elevator"
    getPosition={[Function]}
    leaveAnimation="elevator"
    maintainContainerHeight={false}
    staggerDelayBy={0}
    staggerDurationBy={0}
    typeName="div"
    verticalAlignment="top"
  >
    <div
      className="col-md-4"
      key="Foo"
    >
      <QualityPanel
        quality={
          Object {
            "color": "#333",
            "description": "some description",
            "name": "Foo",
            "score": 50,
          }
        }
      />
    </div>
    <div
      className="col-md-4"
      key="Bar"
    >
      <QualityPanel
        quality={
          Object {
            "color": "#AAA",
            "description": "other description",
            "name": "Bar",
            "score": -25,
          }
        }
      />
    </div>
  </FlipMovePropConverter>
</styled.section>
`;
